custom-routes-vis/src/main/webapp/script.js

```diff
@@ -143,9 +160,174 @@ function hideMarker(markerName) {
+ * @param {!Object} routeJson JSON object containing all information of the target
+ * route.
+ * @param {boolean} directionsApi Whether the route is obtained by the Directions
+ * API; if false, assume it is obtained by the Routes Preferred API.
+ */
+function createRoutePolyline(routeNum, routeJson, directionsApi) {
+  let routeLegs = routeJson.legs;
+  let routeCoordinates = [];
+
+  // Total duration of route in seconds.
+  let totalRouteDurationSec = directionsApi ? 0 : routeJson.duration;
+  // Total distance of route in meters.
+  let totalDistanceMeters = directionsApi ? 0 : routeJson.distanceMeters;
+  // Note: Routes Preferred has duration and distanceMeters attributes for each
+  // route, but Directions only has them for each leg of the route.
+
+  // Get all coordinates given by the steps of the legs of the route.
+  for (let i = 0; i < routeLegs.length; i++) {
+    let legSteps = routeLegs[i].steps;
+    console.log('num steps', legSteps.length);
+
+    for (let j = 0; j < legSteps.length; j++) {
+      directionsApi ?
+          routeCoordinates.push(legSteps[j].start_location) :
+          routeCoordinates.push(legSteps[j].startLocation.LatLng);
+    }
+
+    // Add the end location to the coordinates array.
+    if (i === routeLegs.length - 1) {
+      directionsApi ?
+          routeCoordinates.push(legSteps[legSteps.length - 1].end_location) :
+          routeCoordinates.push(legSteps[legSteps.length - 1].endLocation.LatLng);
+    }
+
+    if (directionsApi) {
+      // Accumulate duration and distance because the total is not directly
+      // available in results from the Directions API.
+      totalRouteDurationSec += parseInt(routeLegs[i].duration.value);
+      totalDistanceMeters += parseInt(routeLegs[i].distance.value);
+    }
+  }
+
+  createRouteFromCoordinates(routeCoordinates, routeNum, totalRouteDurationSec,
+      totalDistanceMeters);
+}
+
+/**
+ * Creates the route on the map and stores it in the routes array.
+ * @param {Array} routeCoordinates The coordinates that the route goes through.
+ * @param {num} routeNum The index of the selected route in the routes array.
+ * @param {num} totalDuration The duration of the route in seconds.
+ * @param {num} totalDistance The distance of the route in meters.
+ */
+function createRouteFromCoordinates(
+    routeCoordinates, 
+    routeNum, 
+    totalDuration, 
+    totalDistance) {
+  let route = new google.maps.Polyline({
+    path: routeCoordinates,
+    geodesic: true,
+    strokeColor: ROUTE_COLORS[routeNum % 3],
+    strokeOpacity: 0.3,
+    strokeWeight: 6,
+  });
+  route.setMap(map);
+  displayedRoutes.push(route);
+
+  // Select the first route as default.
+  if (routeNum === 0) {
+    selectRouteDisplayDetails(0, totalDuration, totalDistance);
+  }
+
+  route.addListener('click', function(event) {
+    selectRouteDisplayDetails(routeNum, totalDuration, totalDistance);
+  });
+}
+
+/** 
+ * Highlights the selected route and displays its duration and distance.
```

Comment:
I would add a mention of the fact you are displaying the distance and duration in hours and miles formats, considering you tell us that you are taking the parameters in seconds and meters format.

Reply:
You didn't address this comment.

Reply:
It there now. Didn't expect you to be so fast lol